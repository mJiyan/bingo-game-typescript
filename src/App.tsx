import React, { useState, useEffect } from 'react';
import useWindowSize from 'react-use/lib/useWindowSize';
import Confetti from 'react-confetti';
import Bingo from './Components/Bingo';

type BingoType = {
  id: number;
  information: string;
  isSelected: boolean;
};

type App = {
  data: Array<BingoType>;
  test: boolean
}

type BingoList = {
  data: Array<BingoType>;
};

// bingo informations of x, y and diagnol axises
const xAxisBingo = [false, false, false, false, false];
const yAxisBingo = [false, false, false, false, false];
let diagnolPositiveBingo = false;
let diagnolNegativeBingo = false;

const App: React.FC<App> = ({ data, test }) => {
  const [bingoData, setBingoData] = useState<Array<BingoType>>(data);
  const [celebration, setCelebration] = useState<boolean>(false);

  const { width, height } = useWindowSize();

  useEffect(() => {
    shuffleBingoData(bingoData);
  }, []);
  

  // it changing state of selected index
  const selectSlot = (id: number): void => {
    bingoData[id].isSelected = !bingoData[id].isSelected;
    checkBingo(id);
    setBingoData((prevData) => [...prevData]);
  };

  // this function find the location (x, y, and diagnol locations) of selected index
  // and returns x-axis, y-axis indexes and also if it placed on diagnol axis or not
  const findTheLocationOfIndex = (index: number) => {
    const xAxis = Math.floor(index / 5);
    const yAxis = index % 5;

    const isDiagnolPositiveAxis = index % 4 === 0;
    const isDiagnolNegativeAxis = index % 6 === 0;

    return { xAxis, yAxis, isDiagnolPositiveAxis, isDiagnolNegativeAxis };
  };

  // this function checks the if there is any bingo
  const checkBingo = (index: number): void => {
    const { xAxis, yAxis, isDiagnolPositiveAxis, isDiagnolNegativeAxis } =
      findTheLocationOfIndex(index);

    const xStartingIndex = xAxis * 5;
    const yStartingIndex = yAxis;

    let isXBingo = true;
    let isYBingo = true;

    // if index is placed on the diagnol axis the function is executed
    // and checks if there is bingo at diagnol axis
    if (isDiagnolPositiveAxis) {
      diagnolPositiveBingo = true;
      for (let q = 0; q < 5; q += 1) {
        if (bingoData[4 + q * 4].isSelected === false && 4 + q * 4 !== 12) {
          diagnolPositiveBingo = false;
        }
      }
    }

    // if index is placed on the diagnol axis the function is executed
    // and checks if there is bingo at diagnol axis
    if (isDiagnolNegativeAxis) {
      diagnolNegativeBingo = true;
      for (let q = 0; q < 5; q += 1) {
        if (bingoData[0 + q * 6].isSelected === false && 0 + q * 6 !== 12) {
          diagnolNegativeBingo = false;
        }
      }
    }

    // checks if there is bingo x-axis and/or y-axis
    for (let i = 0; i < 5; i += 1) {
      if (bingoData[xStartingIndex + i].isSelected === false && xStartingIndex + i !== 12) {
        isXBingo = false;
      }
      if (bingoData[yStartingIndex + i * 5].isSelected === false && yStartingIndex + i * 5 !== 12) {
        isYBingo = false;
      }
    }

    // if there is a bingo than the celebration animation will execute
    if ((isXBingo || isYBingo || diagnolPositiveBingo || diagnolNegativeBingo) && !test) {
      setCelebration(true);
      setTimeout(() => setCelebration(false), 5000);
    }

    // it sets bingo values for rows
    if (isXBingo) xAxisBingo[xAxis] = true;
    else xAxisBingo[xAxis] = false;

    // it sets bingo values for columns
    if (isYBingo) yAxisBingo[yAxis] = true;
    else yAxisBingo[yAxis] = false;
  };

  // it returns bingo for rows
  const xBingo = (index: number): boolean => xAxisBingo[Math.floor(index / 5)];

  // it returns bingo for cols
  const yBingo = (index: number): boolean => yAxisBingo[index % 5];

  // it returns bingo for one of the diagnol axis
  const positiveDiagnolBingo = (index: number): boolean =>
    index % 4 === 0 && index !== 0 && index !== 24;

  // it returns bingo for one of the diagnol axis
  const negativeDiagnolBingo = (index: number): boolean => index % 6 === 0;

  // it changes selected slot styles
  const isItSelected = (isSelected: boolean): string =>
    isSelected === true ? 'opacity-20 line-through ' : '';

  // it randomly sorting bingo card for each users
  const shuffleBingoData = (array: Array<BingoType>): void => {
    array.sort(() => Math.random() - 0.5);
    setBingoData((prevData) => [...prevData]);
  };

  return (
    <>
      {!test && celebration && <Confetti width={width} height={height} recycle={false} />}
      <div className="bg-gradient-to-r from-mainColor to-secondColor">
        <div className="flex flex-col items-center justify-center h-screen -rotate-2">
          <div className="grid grid-cols-5 grid-rows-6">
            <Bingo />
            {bingoData &&
              bingoData.map(
                (item, index) =>
                  (index !== 12 && (
                    <div className="border border-black p-2 sm:p-1 bg-white" key={index}>
                      <div
                        className={`flex flex-col h-32 w-32 md:h-32 md:w-32 sm:w-14 sm:h-14 ${
                          (yBingo(index) ||
                            xBingo(index) ||
                            (diagnolPositiveBingo && positiveDiagnolBingo(index)) ||
                            (diagnolNegativeBingo && negativeDiagnolBingo(index))) &&
                          'bg-mainColor'
                        } `}
                        data-testid={`test-box-${index}`}
                      >
                        <header
                          className="flex flex-col items-end"
                          data-testid={`test-header-${index}`}
                        >
                          <p
                            className={`text-sm md:text-sm sm:text-xs lg:p-1 md:p-1 lg:mt-1 md:p-1 fixed ${
                              item.isSelected && 'opacity-20'
                            }`}
                          >
                            {index}
                          </p>
                        </header>
                        <div
                          className="flex flex-row items-center justify-center cursor-pointer h-screen w-full px-2"
                          onClick={() => selectSlot(index)}
                          data-testid={`test-button-${index}`}
                        >
                          <p
                            className={`lg:text-base md:text-sm sm:text-xs ${isItSelected(
                              item.isSelected,
                            )}`}
                            data-testid={`test-content-${index}`}
                          >
                            {item.information}
                          </p>
                        </div>
                      </div>
                    </div>
                  )) ||
                  (index === 12 && (
                    <div className="border border-black p-2 sm:p-1 bg-white" key={12}>
                      <div className="flex flex-col h-32 w-32 md:h-32 md:w-32 sm:w-14 sm:h-14 border-4 sm:border-2 border-black rounded-full">
                        <header className="flex flex-col items-end ">
                          <p className="text-sm md:text-sm sm:text-xs lg:p-1 md:p-1 lg:mt-1 md:p-1 fixed" />
                        </header>
                        <div className="flex flex-row items-center justify-center cursor-pointer h-screen w-full px-2 ">
                          <p className="lg:text-base md:text-sm sm:text-xs">CONF CALL ðŸ˜· BINGO</p>
                        </div>
                      </div>
                    </div>
                  )),
              )}
          </div>
        </div>
      </div>
    </>
  );
};

export default App;
