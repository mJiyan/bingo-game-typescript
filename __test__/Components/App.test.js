import React from 'react';
import App from '../../src/App';
import { mount } from 'enzyme';
import 'jsdom-global/register'; //at the top of file , even  , before importing react
import toJson from 'enzyme-to-json'; //added this line

const wrapper = mount(<App />);

describe('rendering components', () => {
  test('it should match with the snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
