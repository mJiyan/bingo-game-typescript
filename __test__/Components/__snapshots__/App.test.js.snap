// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering components it should match with the snapshot 1`] = `
<App>
  <div
    className="bg-gradient-to-r from-mainColor to-secondColor"
  >
    <div
      className="flex flex-col items-center justify-center h-screen -rotate-2"
    >
      <div
        className="grid grid-cols-5 grid-rows-6"
      >
        <Bingo>
          <div
            className="border border-black p-2 sm:p-1 bg-white"
          >
            <div
              className="flex flex-col h-32 w-32 md:h-32 md:w-32 sm:w-14 sm:h-14 border-4 sm:border-2 border-black bg-black"
            >
              <header
                className="flex flex-col items-end "
              >
                <p
                  className="text-sm md:text-sm sm:text-xs lg:p-1 md:p-1 lg:mt-1 md:p-1 fixed"
                />
              </header>
              <div
                className="flex flex-row items-center justify-center cursor-pointer h-screen w-full px-2 "
              >
                <p
                  className="lg:text-base md:text-sm sm:text-xs text-white"
                >
                  B
                </p>
              </div>
            </div>
          </div>
          <div
            className="border border-black p-2 sm:p-1 bg-white"
          >
            <div
              className="flex flex-col h-32 w-32 md:h-32 md:w-32 sm:w-14 sm:h-14 border-4 sm:border-2 border-black bg-black"
            >
              <header
                className="flex flex-col items-end "
              >
                <p
                  className="text-sm md:text-sm sm:text-xs lg:p-1 md:p-1 lg:mt-1 md:p-1 fixed"
                />
              </header>
              <div
                className="flex flex-row items-center justify-center cursor-pointer h-screen w-full px-2 "
              >
                <p
                  className="lg:text-base md:text-sm sm:text-xs text-white"
                >
                  I
                </p>
              </div>
            </div>
          </div>
          <div
            className="border border-black p-2 sm:p-1 bg-white"
          >
            <div
              className="flex flex-col h-32 w-32 md:h-32 md:w-32 sm:w-14 sm:h-14 border-4 sm:border-2 border-black bg-black"
            >
              <header
                className="flex flex-col items-end "
              >
                <p
                  className="text-sm md:text-sm sm:text-xs lg:p-1 md:p-1 lg:mt-1 md:p-1 fixed"
                />
              </header>
              <div
                className="flex flex-row items-center justify-center cursor-pointer h-screen w-full px-2 "
              >
                <p
                  className="lg:text-base md:text-sm sm:text-xs text-white"
                >
                  N
                </p>
              </div>
            </div>
          </div>
          <div
            className="border border-black p-2 sm:p-1 bg-white"
          >
            <div
              className="flex flex-col h-32 w-32 md:h-32 md:w-32 sm:w-14 sm:h-14 border-4 sm:border-2 border-black bg-black"
            >
              <header
                className="flex flex-col items-end "
              >
                <p
                  className="text-sm md:text-sm sm:text-xs lg:p-1 md:p-1 lg:mt-1 md:p-1 fixed"
                />
              </header>
              <div
                className="flex flex-row items-center justify-center cursor-pointer h-screen w-full px-2 "
              >
                <p
                  className="lg:text-base md:text-sm sm:text-xs text-white"
                >
                  G
                </p>
              </div>
            </div>
          </div>
          <div
            className="border border-black p-2 sm:p-1 bg-white"
          >
            <div
              className="flex flex-col h-32 w-32 md:h-32 md:w-32 sm:w-14 sm:h-14 border-4 sm:border-2 border-black bg-black"
            >
              <header
                className="flex flex-col items-end "
              >
                <p
                  className="text-sm md:text-sm sm:text-xs lg:p-1 md:p-1 lg:mt-1 md:p-1 fixed"
                />
              </header>
              <div
                className="flex flex-row items-center justify-center cursor-pointer h-screen w-full px-2 "
              >
                <p
                  className="lg:text-base md:text-sm sm:text-xs text-white"
                >
                  O
                </p>
              </div>
            </div>
          </div>
        </Bingo>
        <div
          className="border border-black p-2 sm:p-1 bg-white"
          key="0"
        >
          <div
            className="flex flex-col h-32 w-32 md:h-32 md:w-32 sm:w-14 sm:h-14 false "
          >
            <header
              className="flex flex-col items-end"
            >
              <p
                className="text-sm md:text-sm sm:text-xs lg:p-1 md:p-1 lg:mt-1 md:p-1 fixed false"
              >
                0
              </p>
            </header>
            <div
              className="flex flex-row items-center justify-center cursor-pointer h-screen w-full px-2"
              onClick={[Function]}
            >
              <p
                className="lg:text-base md:text-sm sm:text-xs "
              >
                You will send the minutes?
              </p>
            </div>
          </div>
        </div>
        <div
          className="border border-black p-2 sm:p-1 bg-white"
          key="1"
        >
          <div
            className="flex flex-col h-32 w-32 md:h-32 md:w-32 sm:w-14 sm:h-14 false "
          >
            <header
              className="flex flex-col items-end"
            >
              <p
                className="text-sm md:text-sm sm:text-xs lg:p-1 md:p-1 lg:mt-1 md:p-1 fixed false"
              >
                1
              </p>
            </header>
            <div
              className="flex flex-row items-center justify-center cursor-pointer h-screen w-full px-2"
              onClick={[Function]}
            >
              <p
                className="lg:text-base md:text-sm sm:text-xs "
              >
                i'll have to get back to you
              </p>
            </div>
          </div>
        </div>
        <div
          className="border border-black p-2 sm:p-1 bg-white"
          key="2"
        >
          <div
            className="flex flex-col h-32 w-32 md:h-32 md:w-32 sm:w-14 sm:h-14 false "
          >
            <header
              className="flex flex-col items-end"
            >
              <p
                className="text-sm md:text-sm sm:text-xs lg:p-1 md:p-1 lg:mt-1 md:p-1 fixed false"
              >
                2
              </p>
            </header>
            <div
              className="flex flex-row items-center justify-center cursor-pointer h-screen w-full px-2"
              onClick={[Function]}
            >
              <p
                className="lg:text-base md:text-sm sm:text-xs "
              >
                can everyone go on mute
              </p>
            </div>
          </div>
        </div>
        <div
          className="border border-black p-2 sm:p-1 bg-white"
          key="3"
        >
          <div
            className="flex flex-col h-32 w-32 md:h-32 md:w-32 sm:w-14 sm:h-14 false "
          >
            <header
              className="flex flex-col items-end"
            >
              <p
                className="text-sm md:text-sm sm:text-xs lg:p-1 md:p-1 lg:mt-1 md:p-1 fixed false"
              >
                3
              </p>
            </header>
            <div
              className="flex flex-row items-center justify-center cursor-pointer h-screen w-full px-2"
              onClick={[Function]}
            >
              <p
                className="lg:text-base md:text-sm sm:text-xs "
              >
                can we take this offline?
              </p>
            </div>
          </div>
        </div>
        <div
          className="border border-black p-2 sm:p-1 bg-white"
          key="4"
        >
          <div
            className="flex flex-col h-32 w-32 md:h-32 md:w-32 sm:w-14 sm:h-14 false "
          >
            <header
              className="flex flex-col items-end"
            >
              <p
                className="text-sm md:text-sm sm:text-xs lg:p-1 md:p-1 lg:mt-1 md:p-1 fixed false"
              >
                4
              </p>
            </header>
            <div
              className="flex flex-row items-center justify-center cursor-pointer h-screen w-full px-2"
              onClick={[Function]}
            >
              <p
                className="lg:text-base md:text-sm sm:text-xs "
              >
                sorry, someting __ with my calendar
              </p>
            </div>
          </div>
        </div>
        <div
          className="border border-black p-2 sm:p-1 bg-white"
          key="5"
        >
          <div
            className="flex flex-col h-32 w-32 md:h-32 md:w-32 sm:w-14 sm:h-14 false "
          >
            <header
              className="flex flex-col items-end"
            >
              <p
                className="text-sm md:text-sm sm:text-xs lg:p-1 md:p-1 lg:mt-1 md:p-1 fixed false"
              >
                5
              </p>
            </header>
            <div
              className="flex flex-row items-center justify-center cursor-pointer h-screen w-full px-2"
              onClick={[Function]}
            >
              <p
                className="lg:text-base md:text-sm sm:text-xs "
              >
                sorry, i was on mute.
              </p>
            </div>
          </div>
        </div>
        <div
          className="border border-black p-2 sm:p-1 bg-white"
          key="6"
        >
          <div
            className="flex flex-col h-32 w-32 md:h-32 md:w-32 sm:w-14 sm:h-14 false "
          >
            <header
              className="flex flex-col items-end"
            >
              <p
                className="text-sm md:text-sm sm:text-xs lg:p-1 md:p-1 lg:mt-1 md:p-1 fixed false"
              >
                6
              </p>
            </header>
            <div
              className="flex flex-row items-center justify-center cursor-pointer h-screen w-full px-2"
              onClick={[Function]}
            >
              <p
                className="lg:text-base md:text-sm sm:text-xs "
              >
                did you ask something
              </p>
            </div>
          </div>
        </div>
        <div
          className="border border-black p-2 sm:p-1 bg-white"
          key="7"
        >
          <div
            className="flex flex-col h-32 w-32 md:h-32 md:w-32 sm:w-14 sm:h-14 false "
          >
            <header
              className="flex flex-col items-end"
            >
              <p
                className="text-sm md:text-sm sm:text-xs lg:p-1 md:p-1 lg:mt-1 md:p-1 fixed false"
              >
                7
              </p>
            </header>
            <div
              className="flex flex-row items-center justify-center cursor-pointer h-screen w-full px-2"
              onClick={[Function]}
            >
              <p
                className="lg:text-base md:text-sm sm:text-xs "
              >
                can you email that to everyone?
              </p>
            </div>
          </div>
        </div>
        <div
          className="border border-black p-2 sm:p-1 bg-white"
          key="8"
        >
          <div
            className="flex flex-col h-32 w-32 md:h-32 md:w-32 sm:w-14 sm:h-14 false "
          >
            <header
              className="flex flex-col items-end"
            >
              <p
                className="text-sm md:text-sm sm:text-xs lg:p-1 md:p-1 lg:mt-1 md:p-1 fixed false"
              >
                8
              </p>
            </header>
            <div
              className="flex flex-row items-center justify-center cursor-pointer h-screen w-full px-2"
              onClick={[Function]}
            >
              <p
                className="lg:text-base md:text-sm sm:text-xs "
              >
                (load painful echo / feedback)
              </p>
            </div>
          </div>
        </div>
        <div
          className="border border-black p-2 sm:p-1 bg-white"
          key="9"
        >
          <div
            className="flex flex-col h-32 w-32 md:h-32 md:w-32 sm:w-14 sm:h-14 false "
          >
            <header
              className="flex flex-col items-end"
            >
              <p
                className="text-sm md:text-sm sm:text-xs lg:p-1 md:p-1 lg:mt-1 md:p-1 fixed false"
              >
                9
              </p>
            </header>
            <div
              className="flex flex-row items-center justify-center cursor-pointer h-screen w-full px-2"
              onClick={[Function]}
            >
              <p
                className="lg:text-base md:text-sm sm:text-xs "
              >
                can somebody grant presenter rights?
              </p>
            </div>
          </div>
        </div>
        <div
          className="border border-black p-2 sm:p-1 bg-white"
          key="10"
        >
          <div
            className="flex flex-col h-32 w-32 md:h-32 md:w-32 sm:w-14 sm:h-14 false "
          >
            <header
              className="flex flex-col items-end"
            >
              <p
                className="text-sm md:text-sm sm:text-xs lg:p-1 md:p-1 lg:mt-1 md:p-1 fixed false"
              >
                10
              </p>
            </header>
            <div
              className="flex flex-row items-center justify-center cursor-pointer h-screen w-full px-2"
              onClick={[Function]}
            >
              <p
                className="lg:text-base md:text-sm sm:text-xs "
              >
                Hello, hello?
              </p>
            </div>
          </div>
        </div>
        <div
          className="border border-black p-2 sm:p-1 bg-white"
          key="11"
        >
          <div
            className="flex flex-col h-32 w-32 md:h-32 md:w-32 sm:w-14 sm:h-14 false "
          >
            <header
              className="flex flex-col items-end"
            >
              <p
                className="text-sm md:text-sm sm:text-xs lg:p-1 md:p-1 lg:mt-1 md:p-1 fixed false"
              >
                11
              </p>
            </header>
            <div
              className="flex flex-row items-center justify-center cursor-pointer h-screen w-full px-2"
              onClick={[Function]}
            >
              <p
                className="lg:text-base md:text-sm sm:text-xs "
              >
                who just joined?
              </p>
            </div>
          </div>
        </div>
        <div
          className="border border-black p-2 sm:p-1 bg-white"
          key="12"
        >
          <div
            className="flex flex-col h-32 w-32 md:h-32 md:w-32 sm:w-14 sm:h-14 border-4 sm:border-2 border-black rounded-full"
          >
            <header
              className="flex flex-col items-end "
            >
              <p
                className="text-sm md:text-sm sm:text-xs lg:p-1 md:p-1 lg:mt-1 md:p-1 fixed"
              />
            </header>
            <div
              className="flex flex-row items-center justify-center cursor-pointer h-screen w-full px-2 "
            >
              <p
                className="lg:text-base md:text-sm sm:text-xs"
              >
                CONF CALL 😷 BINGO
              </p>
            </div>
          </div>
        </div>
        <div
          className="border border-black p-2 sm:p-1 bg-white"
          key="13"
        >
          <div
            className="flex flex-col h-32 w-32 md:h-32 md:w-32 sm:w-14 sm:h-14 false "
          >
            <header
              className="flex flex-col items-end"
            >
              <p
                className="text-sm md:text-sm sm:text-xs lg:p-1 md:p-1 lg:mt-1 md:p-1 fixed false"
              >
                13
              </p>
            </header>
            <div
              className="flex flex-row items-center justify-center cursor-pointer h-screen w-full px-2"
              onClick={[Function]}
            >
              <p
                className="lg:text-base md:text-sm sm:text-xs "
              >
                (Child noises in the background)
              </p>
            </div>
          </div>
        </div>
        <div
          className="border border-black p-2 sm:p-1 bg-white"
          key="14"
        >
          <div
            className="flex flex-col h-32 w-32 md:h-32 md:w-32 sm:w-14 sm:h-14 false "
          >
            <header
              className="flex flex-col items-end"
            >
              <p
                className="text-sm md:text-sm sm:text-xs lg:p-1 md:p-1 lg:mt-1 md:p-1 fixed false"
              >
                14
              </p>
            </header>
            <div
              className="flex flex-row items-center justify-center cursor-pointer h-screen w-full px-2"
              onClick={[Function]}
            >
              <p
                className="lg:text-base md:text-sm sm:text-xs "
              >
                i was having connection issues
              </p>
            </div>
          </div>
        </div>
        <div
          className="border border-black p-2 sm:p-1 bg-white"
          key="15"
        >
          <div
            className="flex flex-col h-32 w-32 md:h-32 md:w-32 sm:w-14 sm:h-14 false "
          >
            <header
              className="flex flex-col items-end"
            >
              <p
                className="text-sm md:text-sm sm:text-xs lg:p-1 md:p-1 lg:mt-1 md:p-1 fixed false"
              >
                15
              </p>
            </header>
            <div
              className="flex flex-row items-center justify-center cursor-pointer h-screen w-full px-2"
              onClick={[Function]}
            >
              <p
                className="lg:text-base md:text-sm sm:text-xs "
              >
                can you repeat, please?
              </p>
            </div>
          </div>
        </div>
        <div
          className="border border-black p-2 sm:p-1 bg-white"
          key="16"
        >
          <div
            className="flex flex-col h-32 w-32 md:h-32 md:w-32 sm:w-14 sm:h-14 false "
          >
            <header
              className="flex flex-col items-end"
            >
              <p
                className="text-sm md:text-sm sm:text-xs lg:p-1 md:p-1 lg:mt-1 md:p-1 fixed false"
              >
                16
              </p>
            </header>
            <div
              className="flex flex-row items-center justify-center cursor-pointer h-screen w-full px-2"
              onClick={[Function]}
            >
              <p
                className="lg:text-base md:text-sm sm:text-xs "
              >
                do you see my screen?
              </p>
            </div>
          </div>
        </div>
        <div
          className="border border-black p-2 sm:p-1 bg-white"
          key="17"
        >
          <div
            className="flex flex-col h-32 w-32 md:h-32 md:w-32 sm:w-14 sm:h-14 false "
          >
            <header
              className="flex flex-col items-end"
            >
              <p
                className="text-sm md:text-sm sm:text-xs lg:p-1 md:p-1 lg:mt-1 md:p-1 fixed false"
              >
                17
              </p>
            </header>
            <div
              className="flex flex-row items-center justify-center cursor-pointer h-screen w-full px-2"
              onClick={[Function]}
            >
              <p
                className="lg:text-base md:text-sm sm:text-xs "
              >
                sorry, i had problems loging in
              </p>
            </div>
          </div>
        </div>
        <div
          className="border border-black p-2 sm:p-1 bg-white"
          key="18"
        >
          <div
            className="flex flex-col h-32 w-32 md:h-32 md:w-32 sm:w-14 sm:h-14 false "
          >
            <header
              className="flex flex-col items-end"
            >
              <p
                className="text-sm md:text-sm sm:text-xs lg:p-1 md:p-1 lg:mt-1 md:p-1 fixed false"
              >
                18
              </p>
            </header>
            <div
              className="flex flex-row items-center justify-center cursor-pointer h-screen w-full px-2"
              onClick={[Function]}
            >
              <p
                className="lg:text-base md:text-sm sm:text-xs "
              >
                i need to jump in another call
              </p>
            </div>
          </div>
        </div>
        <div
          className="border border-black p-2 sm:p-1 bg-white"
          key="19"
        >
          <div
            className="flex flex-col h-32 w-32 md:h-32 md:w-32 sm:w-14 sm:h-14 false "
          >
            <header
              className="flex flex-col items-end"
            >
              <p
                className="text-sm md:text-sm sm:text-xs lg:p-1 md:p-1 lg:mt-1 md:p-1 fixed false"
              >
                19
              </p>
            </header>
            <div
              className="flex flex-row items-center justify-center cursor-pointer h-screen w-full px-2"
              onClick={[Function]}
            >
              <p
                className="lg:text-base md:text-sm sm:text-xs "
              >
                lets wait for __!
              </p>
            </div>
          </div>
        </div>
        <div
          className="border border-black p-2 sm:p-1 bg-white"
          key="20"
        >
          <div
            className="flex flex-col h-32 w-32 md:h-32 md:w-32 sm:w-14 sm:h-14 false "
          >
            <header
              className="flex flex-col items-end"
            >
              <p
                className="text-sm md:text-sm sm:text-xs lg:p-1 md:p-1 lg:mt-1 md:p-1 fixed false"
              >
                20
              </p>
            </header>
            <div
              className="flex flex-row items-center justify-center cursor-pointer h-screen w-full px-2"
              onClick={[Function]}
            >
              <p
                className="lg:text-base md:text-sm sm:text-xs "
              >
                Next slide, please.
              </p>
            </div>
          </div>
        </div>
        <div
          className="border border-black p-2 sm:p-1 bg-white"
          key="21"
        >
          <div
            className="flex flex-col h-32 w-32 md:h-32 md:w-32 sm:w-14 sm:h-14 false "
          >
            <header
              className="flex flex-col items-end"
            >
              <p
                className="text-sm md:text-sm sm:text-xs lg:p-1 md:p-1 lg:mt-1 md:p-1 fixed false"
              >
                21
              </p>
            </header>
            <div
              className="flex flex-row items-center justify-center cursor-pointer h-screen w-full px-2"
              onClick={[Function]}
            >
              <p
                className="lg:text-base md:text-sm sm:text-xs "
              >
                is ___ on the call?
              </p>
            </div>
          </div>
        </div>
        <div
          className="border border-black p-2 sm:p-1 bg-white"
          key="22"
        >
          <div
            className="flex flex-col h-32 w-32 md:h-32 md:w-32 sm:w-14 sm:h-14 false "
          >
            <header
              className="flex flex-col items-end"
            >
              <p
                className="text-sm md:text-sm sm:text-xs lg:p-1 md:p-1 lg:mt-1 md:p-1 fixed false"
              >
                22
              </p>
            </header>
            <div
              className="flex flex-row items-center justify-center cursor-pointer h-screen w-full px-2"
              onClick={[Function]}
            >
              <p
                className="lg:text-base md:text-sm sm:text-xs "
              >
                sorry, i didn't found the conference id
              </p>
            </div>
          </div>
        </div>
        <div
          className="border border-black p-2 sm:p-1 bg-white"
          key="23"
        >
          <div
            className="flex flex-col h-32 w-32 md:h-32 md:w-32 sm:w-14 sm:h-14 false "
          >
            <header
              className="flex flex-col items-end"
            >
              <p
                className="text-sm md:text-sm sm:text-xs lg:p-1 md:p-1 lg:mt-1 md:p-1 fixed false"
              >
                23
              </p>
            </header>
            <div
              className="flex flex-row items-center justify-center cursor-pointer h-screen w-full px-2"
              onClick={[Function]}
            >
              <p
                className="lg:text-base md:text-sm sm:text-xs "
              >
                (animal noises in the background)
              </p>
            </div>
          </div>
        </div>
        <div
          className="border border-black p-2 sm:p-1 bg-white"
          key="24"
        >
          <div
            className="flex flex-col h-32 w-32 md:h-32 md:w-32 sm:w-14 sm:h-14 false "
          >
            <header
              className="flex flex-col items-end"
            >
              <p
                className="text-sm md:text-sm sm:text-xs lg:p-1 md:p-1 lg:mt-1 md:p-1 fixed false"
              >
                24
              </p>
            </header>
            <div
              className="flex flex-row items-center justify-center cursor-pointer h-screen w-full px-2"
              onClick={[Function]}
            >
              <p
                className="lg:text-base md:text-sm sm:text-xs "
              >
                Could you share this slides afterwards?
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</App>
`;
